### DISCUSSION TITLE
"jest" is not defined with "@storybook/addon-storyshots"

### DISCUSSION THREAD

**Discussion by jackdh** [vitest-dev/vitest#1514] [Q&A]
Hi,

I'm trying to use `vite` & `vitest` with Storybook's image storyshots.

I can get pupperteer to work however when I try and use: `import initStoryshots from "@storybook/addon-storyshots";`

I get the following error:

```
ReferenceError: jest is not defined
 ❯ Object.load node_modules/@storybook/addon-storyshots/dist/ts3.9/frameworks/react/loader.js:12:23
 ❯ Object.loadFramework [as default] node_modules/@storybook/addon-storyshots/dist/ts3.9/frameworks/frameworkLoader.js:26:19
 ❯ Proxy.testStorySnapshots node_modules/@storybook/addon-storyshots/dist/ts3.9/api/index.js:28:94
 ❯ src/stories/image-snapshots.test.ts:6:15
      4| 
      5| import initStoryshots from "@storybook/addon-storyshots";
      6| initStoryshots();
```

I've enabled `golbals: true` in the `vite.config.ts`.

Is this a vite issue not stubbing filling `jest` or is this a storybook issue for only supporting `jest`?

Thanks!

**Comment by sheremet-va** [comment]
Vitest doesn't provide `jest` variable, since we are not jest, but we have compatible `vi`. You can add it in `setupFiles`: 

```
globalThis.jest = vi
```

**Reply by jackchoumine** [reply]
> Vitestb doesn't provide `jest` variable, since we are not jest, but we have compatible `vi`. You can add it in `setupFiles`:
> 
> ```
> globalThis.jest = vi
> ```

I have same issue, which steupFile, could u giva more info? or some code?

**Reply by sheremet-va** [reply]
https://vitest.dev/config/#setupfiles

Code I already gave

**Comment by jackchoumine** [comment]
> Hi,
> 
> I'm trying to use `vite` & `vitest` with Storybook's image storyshots.
> 
> I can get pupperteer to work however when I try and use: `import initStoryshots from "@storybook/addon-storyshots";`
> 
> I get the following error:
> 
> ```
> ReferenceError: jest is not defined
>  ❯ Object.load node_modules/@storybook/addon-storyshots/dist/ts3.9/frameworks/react/loader.js:12:23
>  ❯ Object.loadFramework [as default] node_modules/@storybook/addon-storyshots/dist/ts3.9/frameworks/frameworkLoader.js:26:19
>  ❯ Proxy.testStorySnapshots node_modules/@storybook/addon-storyshots/dist/ts3.9/api/index.js:28:94
>  ❯ src/stories/image-snapshots.test.ts:6:15
>       4| 
>       5| import initStoryshots from "@storybook/addon-storyshots";
>       6| initStoryshots();
> ```
> 
> I've enabled `golbals: true` in the `vite.config.ts`.
> 
> Is this a vite issue not stubbing filling `jest` or is this a storybook issue for only supporting `jest`?
> 
> Thanks!

I have the same issue, have u get a solution?  what it is?

**Comment by bichotll-gartner** [comment]
I wonder if there is any chance where we could make the jest variable global for those scenarios?
